metadata-ingestion/src/datahub/ingestion/source/usage/snowflake_usage.py

```diff
@@ -335,6 +336,14 @@ def is_object_valid(obj: Dict[str, Any]) -> bool:
                 event_dict["query_start_time"]
             ).astimezone(tz=timezone.utc)
 
+            if not event_dict["email"] and self.config.email_domain:
+                if not event_dict["user_name"]:
+                    event_dict["user_name"] = "unknown"
```

Comment:
If we don't have a username we are falling back to unknown? This seems kinda strange - why not just use None?

Reply:
I changed it to skip users without email address or username